sambaplayersdk/src/main/java/com/sambatech/player/plugins/TrackingLive.java

```diff
@@ -0,0 +1,313 @@
+                sttmLive.trackPlayAndONEvent();
+                sttmLive.startOnEventTask();
+            }
+
+        }
+
+        @Override
+        public void onPause(SambaEvent event) {
+            super.onPause(event);
+            sttmLive.cancelOnEventTask();
+            sttmLive.trackPauseEvent();
+        }
+
+        @Override
+        public void onError(SambaEvent event) {
+            super.onError(event);
+            sttmLive.cancelOnEventTask();
+        }
+    };
+
+
+    public void onLoad(@NonNull SambaPlayer player) {
+        Log.i("track", "load");
+
+        this.context = player.getContext();
+        this.media = (SambaMediaConfig) player.getMedia();
+
+        init();
+
+        PluginManager.getInstance().notifyPluginLoaded(this);
+    }
+
+    public void onInternalPlayerCreated(@NonNull SimpleExoPlayerView internalPlayer) {
+    }
+
+    public void onDestroy() {
+        Log.i("track", "destroy");
+        SambaEventBus.unsubscribe(playerListener);
+
+        if (sttmLive != null) {
+            sttmLive.destroy();
+            sttmLive = null;
+        }
+    }
+
+    private void init() {
+        if (media.sttmUrl != null && sttmLive == null) {
+            sttmLive = new SttmLive();
+        }
+
+        if (media.projectHash != null && media.id != null) {
+            SambaEventBus.subscribe(playerListener);
+        }
+    }
+
+    private Sttm2 getSttm2() throws Exception {
+
+        InputStream inputStream = null;
+        Scanner scanner = null;
+
+        try {
+
+            URL myURL = new URL(getSttm2RequestUrl());
+            HttpURLConnection conn = (HttpURLConnection) myURL.openConnection();
+            conn.setRequestProperty("Content-Type", "text/plain");
+
+            int responseCode = conn.getResponseCode();
+            if (responseCode >= 200 && responseCode < 299) {
+                inputStream = conn.getInputStream();
+                scanner = new Scanner(inputStream);
+                return getSttm2ResponseFromScanner(scanner);
+            }
+
+        } finally {
+            try {
+                if (inputStream != null)
+                    inputStream.close();
+
+                if (scanner != null)
+                    scanner.close();
+            } catch (IOException e) {
+                Log.w(getClass().getSimpleName(), "Error closing server request", e);
+            }
+        }
+
+        return null;
+    }
+
+
+    private Sttm2 getSttm2ResponseFromScanner(Scanner scanner) throws Exception {
+
+        int delimiter = media.id != null ? Integer.parseInt(media.id.split("(?=\\d[a-zA-Z]*$)")[1].substring(0, 1)) : 0;
+
+        Scanner scannerDelimited = scanner.useDelimiter("\\A");
+
+        if (scannerDelimited.hasNext()) {
+            String token = scannerDelimited.next();
+
+            token = token.substring(delimiter, token.length() - delimiter).replaceAll("-", "+").replaceAll("_", "/");
+
+            switch (token.length() % 4) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.sambatech.com/coding_rules#rule_key=squid%3AS109)